docs/content/preview/architecture/transactions/concurrency-control.md

```diff
@@ -0,0 +1,1166 @@
+
+Each retry will work a newer snapshot of the database in anticipation that the conflicts might not occur (if an earlier conflicting transaction T2 has committed with a commit time before the read time of the new snapshot, the conflicts with T2 would essentially be voided since T1 and T2 would no longer be "concurrent").
+
+Also note that the retries will not be performed in case the amount of data to be sent from YSQL to the client proxy exceeds the TServer gflag `ysql_output_buffer_size`.
+
+## Wait-on-Conflict [[Beta]](/preview/faq/general/#what-is-the-definition-of-the-beta-feature-tag)
+
+This mode of concurrency control is applicable only for YSQL and provides the same semantics as PostgreSQL.
+
+In this mode, transactions are not assigned priorities. If a conflict occurs when a transaction T1 tries to read/ write/ lock a row in a conflicting mode with a few other concurrent transactions, T1 will **wait** until all conflicting transactions finish by either committing or rolling back. Once all conflicting transactions have finished, T1 will:
+
+1. make progress if the conflicting transactions didn’t commit any permanent modifications that conflict with T1.
+2. abort otherwise.
+
+By default, transactions operate in the priority based `Fail-on-Conflict` mode. This mode can be enabled by setting the YB-TServer gflag `enable_wait_queues=true`, which start in-memory wait queues that provide waiting semantics when conflicts are detected between transactions. A rolling restart is needed for the gflag to take effect.
+
+Since T1 can make progress only if the conflicting transactions didn’t commit any conflicting permanent modifications, there are some intricacies in the behaviour. The list of exhaustive cases possible are detailed below in the Examples section.
+
+{{< note >}}
+
+1. Semantics of [Read Committed](../read-committed/) isolation make sense only with the Wait-on-Conflict behaviour.
+1. Transactions in Read Committed isolation follow `Wait-on-Conflict` policy even without wait queues i.e., even when `enable_wait_queues=false`. This is done by relying on a indefinite retry-backoff mechanism with exponential delays when conflicts are detected. However, when `Read Committed` isolation provides Wait-on-Conflict semantics without wait queues, the following limitations exist -
+    1. there might be a need to manually tune the exponential backoff parameters for performance as explained [here](../read-committed/#performance-tuning).
+    1. the app will have to [rely on statement timeouts to avoid deadlocks](../read-committed/#avoid-deadlocks-in-read-committed-transactions).
+    1. there can be unfairness and high P99 latencies during contention due to the retry-backoff mechanism.
+{{</note >}}
+
+{{< note title="Best-effort internal retries for first statement in a transaction apply even in Wait-on-Conflict policy" >}}
```

Comment:
Since there are two suggestions, changed to this sort of middle path - `Best-effort internal retries also apply for the first statement in transactions following the Wait-on-Conflict policy`, let me know